app/views/confirmations/sent.html.erb

```diff
@@ -30,17 +29,27 @@
   } %>
 <% end %>
 
-<p class="govuk-body" data-module="gem-track-click">
-  <% if service %>
-    <a class="govuk-link" href="<%= service[:url] %>" data-module="explicit-cross-domain-links" data-track-category="account-create" data-track-action="confirm-email" data-track-label="<%= service[:name] %>">
-      <%= t("confirmation_sent.continue_to_service", service_name: service[:name]) %>
-    </a>
-  <% else %>
+<% if @user_is_new && params[:previous_url]&.start_with?(oauth_authorization_path) %>
+  <%= form_tag redirect_to_previous_url_path, method: :post, :"data-module" => "explicit-cross-domain-links" do %>
+    <%= hidden_field_tag :previous_url, params[:previous_url] %>
```

Comment:
I assume `params[:previous_url]` will contain a querystring? 

Did you consider exploding the querystring itself into hidden fields? That would allow you to set the form method to `get` and remove the need for the extra controller.

Reply:
I didn't do that because `explicit-cross-domain-links` also adds a query parameter to the `action` field, and that's done on the frontend, so we can't add a hidden field here... but on second thought, that can be worked around.  I'll change it.

Reply:
Hmm, I think the javascript (in the publishing components) would need to change.

I thought we could add the `data-module` to the form on the *previous* page, so that the user arrives on this page with a `_ga` param, and so we can stick that into a hidden field.  But the form on the previous page already has a different `data-module`.

Reply:
Sounds like it's not worth it just to save us from an extra controller.